### PR TITLE
Editor: Fix `_parse_category()` is not called for custom categories

### LINKED ISSUE #87910: EditorInspectorPlugin not parsing custom category
### Tested versions

- Reproducible in: v4.2.1.stable.official [b09f793f5], v4.3.dev2.official [352434668]
- Not reproducible in: v4.1.3.stable.official

### System information

Godot v4.2.1.stable - Windows 10.0.22631

### Issue description

When creating a plugin for Godot I'm adding a custom button to the editor interface to do something when pressed.

I add this button to a custom category for a specific custom Class. This way I can put the button in the editor interface where I want it to be.

I started the plugin in Godot v4.1, and now when I try to use it in Godot v4.2, the button is not showing. I found out that the button was not showing because I was using `@export_category`.

In the MRP you can find the button in the editor interface as expected in v4.1:

EditorInspectorPluginCustomCategoryIssue413

But in v4.2.1, the button is not showing:

EditorInspectorPluginCustomCategoryIssue421

This is still happening in Godot v4.3-dev2.

The Custom Node script is just this:
```gdscript
class_name CustomNode extends Node

@export var a_variable: String
@export_category("Category")
@export var another_variable: String
```

And the EditorPlugin code is just this:
```gdscript
@tool
extends EditorPlugin

var plugin

func _enter_tree():
	plugin = preload("res://addons/myinspectorplugin/my_inspector_plugin.gd").new()
	add_inspector_plugin(plugin)

func _exit_tree():
	remove_inspector_plugin(plugin)
```

The problem seems to be that the `_parse_category()` of `EditorInspectorPlugin` is not parsing custom categories from scripts.

### Steps to reproduce

1. Open the MRP. (It's built in v4.2.1 so if you open it with v.4.1.3 it will show that it needs to be modified, it doesn't matter. You can just click OK).
2. Open the `custom_node.tscn`.
3. Click the CustomNode in the scene tree.
4. Look at the inspector

In Godot v4.1.3, a button with the name `MyCustomCategoryButton` will appear as expected.
In Godot v4.2.1, that button will not be showing.

### Minimal reproduction project (MRP)

[EditorInspectorPluginCustomCategoryIssue421.zip](https://github.com/godotengine/godot/files/14152591/EditorInspectorPluginCustomCategoryIssue421.zip)

### PR DESCRIPTION
**Pull request by dalexeev**
* Fixes #87910.

### CODE REVIEW

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -2816,53 +2816,53 @@ void EditorInspector::update_tree() {
...
-					if (!docs.is_empty()) {
-						// The documentation of a GDScript's main class is at the end of the array.
-						// Hacky because this isn't necessarily always guaranteed.
-						doc_name = docs[docs.size() - 1].name;
-					}
-					if (script_name != StringName()) {
-						label = script_name;
-					}
+				// Use category's owner script to update some of its information.
+				if (!EditorNode::get_editor_data().is_type_recognized(type) && p.hint_string.length() && ResourceLoader::exists(p.hint_string)) {
```
Comment (kleonc) [inline editor/editor_inspector.cpp:2829]: Nitpicking but it's already guaranteed to be non-empty here.
```suggestion
				if (!EditorNode::get_editor_data().is_type_recognized(type) && ResourceLoader::exists(p.hint_string)) {
```

**Review by kleonc — APPROVED** [review]
Makes sense!

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**